auto1111sdk/pipelines/StableDiffusionPipeline.py

```diff
@@ -322,7 +322,7 @@ def inpainting_img2img(self, mask: Image, init_image: Image, prompt: str, negati
         return output_images
     
     def sd_upscale_img2img(self, prompt: str, init_image: Image, upscaler = None, overlap: int = 64, scale_factor: int = 2, negative_prompt: str = "", seed: int = -1, steps: int = 20, cfg_scale: float = 7.0, num_images: int = 1, sampler_name: str = 'Euler', denoising_strength: float = 0.75):
-        if isinstance(upscaler, (EsrganPipeline, RealEsrganPipeline)):
+        if not isinstance(upscaler, (EsrganPipeline, RealEsrganPipeline)):
             raise ValueError("Upscaler should either be an instance of RealEsrgan pipeline or Esrgan pipeline")
         input_params = { 
             'init_images': [init_image], 
```

Comment:
Btw, because of this line, `sd_upscale_img2img` will throw the error below, as `__process_args_img2img` is expecting `init_image`. 
```
in __process_args_img2img
    raise TypeError("Missing input image")
```

Please fix it, too.